### ISSUE TITLE
macOS Catalina 10.15.7 - process hangs on multiple files moved at the same time

### ISSUE THREAD

**Issue by Russell-Tran** [gorakhargosh/watchdog#736]
Hi there! I've encountered this bug for macOS Catalina `10.15.7` on Python `3.7` and `3.9` on watchdog `1.0.1`. (Installed using pipenv). Using the example code,
```
import sys
import time
import logging
from watchdog.observers import Observer
from watchdog.events import LoggingEventHandler

if __name__ == "__main__":
    logging.basicConfig(level=logging.INFO,
                        format='%(asctime)s - %(message)s',
                        datefmt='%Y-%m-%d %H:%M:%S')
    path = sys.argv[1] if len(sys.argv) > 1 else '.'
    event_handler = LoggingEventHandler()
    observer = Observer()
    observer.schedule(event_handler, path, recursive=True)
    observer.start()
    try:
        while True:
            time.sleep(1)
    finally:
        observer.stop()
        observer.join()
```

If you move a file into or out of the designated directory, it logs the change successfully. However, if you move more than one file into or out of the directory simultaneously, the change is not logged. Moreover, any successive single-file moves into or out of the directory after attempting a multiple-file move will not be logged either, possibly suggesting something is hanging. Any thoughts? Thank you!

**Comment by Russell-Tran** [comment]
I am able to reproduce this on watchdog `1.0.0` as well.

**Comment by Russell-Tran** [comment]
@ceelian @CCP-Aporia in case you had any input on this issue or my other issue (#737). Thanks!

**Comment by CCP-Aporia** [comment]
Thank you for this report! Unfortunately the `FSEventsObserver` has a couple of issues at the moment, but there are a few PRs in the works to resolve all of them.

**Comment by ceelian** [comment]
Am I right that the latest stable version for macos is currently 0.10.3 or did I miss any intermediate stable release after the performance improvement PR? Thanks for the huge effort to improve the performance for macos, the speed is really amazing.

**Comment by Russell-Tran** [comment]
Hi there! The bug for Catalina 10.15.7 seems resolved with watchdog `1.0.2`. Thank you to everyone who worked on it!

**Comment by BoboTiG** [comment]
Thanks for the confirmation @Russell-Tran :)
